src/sage/numerical/pivot_rules_for_simplex_method.py

```diff
@@ -0,0 +1,776 @@
+
+        - ``weight`` -- object convertible to a sage ``vector`` in the ambient space of current_dictionary.
+
+        EXAMPLES::
+
+            sage: from sage.numerical.pivot_rules_for_simplex_method import SimplexMethodPivot
+            sage: A = ([1, 1], [3, 1])
+            sage: b = (1000, 1500)
+            sage: c = (10, 5)
+            sage: P = InteractiveLPProblemStandardForm(A, b, c)
+            sage: D = P.initial_dictionary()
+            sage: D.entering()
+
+            sage: D.leaving()
+
+            sage: def eta(v): # use 2 norm as normalization
+            ....:     return v.norm()
+            sage: pivot = SimplexMethodPivot("NW_rule")
+            sage: pivot(D, False, eta, [1,1])
+            sage: D.entering()
+            x1
+            sage: D.leaving()
+            x4
+
+            Inputs are assumed to be correct for the current problem::
+            sage: D.update()
+            sage: pivot(D, False, eta, 1)
+            Traceback (most recent call last):
+            ...
+            TypeError: 'sage.rings.integer.Integer' object is not iterable
+            sage: eta = [1,2,1]
+            sage: pivot(D, False, eta, [1,1])
+            Traceback (most recent call last):
+            ...
+            TypeError: 'list' object is not callable
+        """
+        eta = eta
+        weight = vector(weight)
+        if current_dictionary.entering() is None:
+            current_vertex = current_dictionary.basic_solution() # This current vertex v of P.
+            best_value = InfinityRing(float('-inf'))
+            best_vars = None
+            for var_in in current_dictionary.possible_entering(): # Explore adjacent vertices to v such that c^tu>c^tv.
+                copy_of_dict = deepcopy(current_dictionary) # TODO: custom copy method for LPAbstractDictionary or better way to explore multiple vertices.
+                copy_of_dict.enter(var_in)
+                for var_out in copy_of_dict.possible_leaving():
+                    copy_of_dict.leave(var_out)
+                    copy_of_dict.update()
+                    next_vertex = copy_of_dict.basic_solution() # Vertex u adjacent to v in gr(P).
+                    if eta(next_vertex - current_vertex) != 0:
+                        current_value = (weight * (next_vertex - current_vertex)) / eta(next_vertex - current_vertex) # NW-rule
+                        if current_value > best_value:
+                            best_vars = var_in, var_out
+                            best_value = current_value
+                    copy_of_dict = deepcopy(current_dictionary)
+            if best_vars is not None: # Don't set entering or leaving variables if we haven't found any strict improvement.
+                current_dictionary.enter(best_vars[0])
+                if current_dictionary.leaving() is None:
+                    current_dictionary.leave(best_vars[1])
+
+
+from sage.numerical.interactive_simplex_method import LPAbstractDictionary
```

Comment:
either move to the `import` block at the top or into the method where it is used